### ISSUE TITLE
Fix S3459 FP: Backing field with `ref` property

### ISSUE THREAD

**Issue by dotjpg3141** [SonarSource/sonar-dotnet#9106]
### Description

False Positive of S3459 for backing field with `ref` property.

### Repro steps
```csharp
public class MyClass
{
    private int _foo;

    public ref int Foo => ref _foo;
}
```

### Expected behavior

No warning

### Actual behavior

warning S3459: Remove unassigned field '_foo', or set its value. (https://rules.sonarsource.com/csharp/RSPEC-3459)

### Known workarounds

Supress this warning manually

### Related information

* C#/VB.NET Plugins version: SonarAnalyzer.CSharp 9.23.2.88755
* Visual Studio version 17.9.2
* MSBuild / dotnet version: dotnet 8.0.200
* SonarScanner for .NET version (if used)
* Operating System: Windows 10

**Comment by CristianAmbrosini** [comment]
Hi @dotjpg3141! I confirm this as a FP, I'll add a reproducer in our codebase shortly. 
Thanks!
